### PR TITLE
update of core storefront packages

### PR DESCRIPTION
**Pull request by sebaholesz**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Some of the core packages used on storefront (Urql, React, Next) were not on their latest versions. This PR updates them to the most recent ones.
|New feature| No
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No
|Fixes issues| ...
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/storefront/pages/_app.tsx [hunk 1]
```diff
@@ -1,12 +1,7 @@
 import 'dayjs/locale/cs';
 import 'dayjs/locale/sk';
-import 'lightgallery/css/lg-thumbnail.css';
-import 'lightgallery/css/lightgallery.css';
-import 'react-loading-skeleton/dist/skeleton.css';
 import '../styles/globals.css';
 import '../styles/user-text.css';
-import 'react-toastify/dist/ReactToastify.css';
-import 'nprogress/nprogress.css';
```
Comment (tvikito) [inline project-base/storefront/pages/_app.tsx:9]: I understand motivation behind this but this is not good step since it's huge [step back for optimalization](https://www.debugbear.com/blog/avoid-css-import). Please revert this change.

File: project-base/storefront/urql/errorExchange.ts [hunk 2]
```diff
@@ -27,10 +31,37 @@ export const getErrorExchange =
...
+const handleCartError = ({ userError, applicationError }: ParsedErrors) => {
+    switch (applicationError?.type) {
+        case ApplicationErrors['cart-not-found']:
+            break;
+        case ApplicationErrors.default:
+            showErrorMessage(applicationError.message, GtmMessageOriginType.cart);
+            break;
+    }
+
+    if (userError?.validation !== undefined) {
```
Comment (tvikito) [inline project-base/storefront/urql/errorExchange.ts:62]: `if (userError?.validation) {`

File: project-base/storefront/hooks/cart/useReloadCart.ts [hunk 3]
```diff
@@ -4,23 +4,31 @@ import { getUrlWithoutGetParameters } from 'helpers/parsing/getUrlWithoutGetPara
...
+    const updateLastVisitedSlug = usePersistStore((store) => store.updateLastVisitedSlug);
+    const lastVisitedSlug = usePersistStore((store) => store.lastVisitedSlug);
     const cartUuid = usePersistStore((store) => store.cartUuid);
+    const client = useClient();
 
     useEffect(() => {
-        if (cartUuid || isUserLoggedIn) {
-            refetchCart();
+        const hasSlugChanged = lastVisitedSlug !== slug;
+        updateLastVisitedSlug(slug);
```
Comment (tvikito) [inline project-base/storefront/hooks/cart/useReloadCart.ts:26]: This seems like something is not done correctly on the way. Cannot this be done better way so it is reacting simply for slug change?

File: project-base/storefront/components/Basic/Link/Link.tsx [hunk 4]
```diff
@@ -15,33 +15,31 @@ type LinkProps = NativePropsAnchor & {
...
+    const props = {
+        className: twMergeCustom(
+            'inline-flex cursor-pointer items-center text-greyDark outline-none hover:text-primary',
+            isButton ? 'no-underline hover:no-underline' : 'underline hover:underline',
+        ),
+        href: isExternal ? href : undefined,
+        rel,
+        target,
+        'data-testid': getDataTestId(isExternal, isButton),
+    };
```
[the remark this answers is not in the document]
Comment (sebaholesz) [reply project-base/storefront/components/Basic/Link/Link.tsx:34]: Done your way. Thanks for pointing this out.

File: project-base/storefront/pages/_app.tsx (hunk 1 above)
Comment (sebaholesz) [reply project-base/storefront/pages/_app.tsx:9]: I did not know! Thanks for pointing it out :confetti_ball:

File: project-base/storefront/urql/errorExchange.ts (hunk 2 above)
Comment (sebaholesz) [reply project-base/storefront/urql/errorExchange.ts:62]: Done

File: project-base/storefront/hooks/cart/useReloadCart.ts (hunk 3 above)
Comment (sebaholesz) [reply project-base/storefront/hooks/cart/useReloadCart.ts]: I think we can. I dont know why it has stayed here.

File: project-base/storefront/components/Basic/Flag/Flag.tsx [hunk 5]
```diff
@@ -12,15 +12,15 @@ type FlagProps = NativeProps & {
...
-            {children}
-        </a>
+    <ExtendedNextLink
+        type="blogCategory"
+        href={href}
+        className={twMergeCustom(
+            'mb-2 mr-3 inline-flex rounded-sm bg-primaryLight py-1 px-2 text-xs uppercase text-dark no-underline hover:text-dark hover:no-underline',
+            classNames,
+        )}
+        data-testid={getDataTestId(dataTestId)}
```
[the remark this answers is not in the document]
Comment (sebaholesz) [reply project-base/storefront/components/Basic/Flag/Flag.tsx]: I changed it to `dataTestId={getDataTestId(dataTestId)}`

[the remark this answers is not in the document]
Comment (tvikito) [reply project-base/storefront/components/Basic/Flag/Flag.tsx]: I am not sure, if you've checked the DOM for this attribute, but I think this will not be enough, or is it? I think you will need to implement it as prop also in `ExtendedNextLink`, like 

```
<NextLink
            as={isStatic ? as : href}
            prefetch={false}
            href={
                isStatic
                    ? href
                    : {
                          pathname: FriendlyPagesDestinations[type],
                          query: { [SLUG_TYPE_QUERY_PARAMETER_NAME]: FriendlyPagesTypes[type], ...queryParams },
                      }
            }
            data-testid={dataTestId}
            {...props}
        >
```

Otherwise it will be just passed to `NextLink` component without being rendered to the DOM as element attribute. But please check it first.

[the remark this answers is not in the document]
Comment (sebaholesz) [reply project-base/storefront/components/Basic/Flag/Flag.tsx]: I actually reverted it to as it was before, meaning
`data-testid={getDataTestId(dataTestId)}`

This way it is included in the DOM.

**Review by tvikito — APPROVED** [review]
